src/native/corehost/hostpolicy/hostpolicy_context.cpp

```diff
@@ -343,6 +343,25 @@ int hostpolicy_context_t::initialize(const hostpolicy_init_t &hostpolicy_init, c
         coreclr_properties.add(common_property::StartUpHooks, startup_hooks.c_str());
     }
 
+    // Path to dotnet corresponding to the runtime used to run the application (if it exists)
+    // Only for framework-dependent applications, as there is no corresponding dotnet for self-contained.
+#if !defined(NATIVE_LIBS_EMBEDDED)
+    if (hostpolicy_init.is_framework_dependent)
+    {
+        pal::string_t dotnet_host_path = hostpolicy_init.host_info.dotnet_root;
+        append_path(&dotnet_host_path, _X("dotnet"));
+        dotnet_host_path.append(_STRINGIFY(EXE_FILE_EXT));
+        if (pal::file_exists(dotnet_host_path))
```

Comment:
It is one more file access during startup that is unnecessary most of the time.

What are the circumstances where the file can be missing?

Reply:
It would pretty much be a broken install or hand constructed - all our installation mechanisms should always have it. I'm fine with putting it on the consumer to check for existence.

Reply:
I think that it would be better.

 I think that the consumer should just fail with FileNotFoundException when the file is missing given that this is most likely a corrupted install.